src/app/TempConverter/TempConverter.c

```diff
@@ -0,0 +1,45 @@
+#include "TempConverter.h"
+
+float TempConverter(float resistance) {
+    float temp[] = {953773.550, 886261.171, 823958.701, 766433.156, 713289.955, 664169.306, 618742.960, 576711.277,
+                    537800.582, 501760.778, 468363.196, 437398.638, 408675.627, 382018.806, 357267.499, 334274.408,
+                    312904.426, 293033.568, 274547.993, 257343.129, 241322.863, 226398.817, 212489.687, 199520.635,
+                    187422.749, 176132.536, 165591.475, 155745.596, 146545.109, 137944.054, 129899.988, 122373.703,
+                    115328.954, 108732.230, 102552.526, 96761.147, 91331.525, 86239.049, 81460.909, 76975.960,
+                    72764.590, 68808.600, 65091.096, 61596.391, 58309.909, 55218.103, 52308.380, 49569.023, 46989.132,
+                    44558.557, 42267.848, 40108.202, 38071.414, 36149.832, 34336.324, 32624.231, 31007.341, 29479.853,
+                    28036.347, 26671.761, 25381.362, 24160.726, 23005.712, 21912.448, 20877.309, 19896.900, 18968.041,
+                    18087.754, 17253.245, 16461.898, 15711.255, 14999.014, 14323.012, 13681.218, 13071.726, 12492.746,
+                    11942.593, 11419.687, 10922.538, 10449.748, 10000.000, 9572.055, 9164.745, 8776.973, 8407.702,
+                    8055.959, 7720.823, 7401.428, 7096.956, 6806.636, 6529.741, 6265.582, 6013.513, 5772.919, 5543.224,
+                    5323.878, 5114.367, 4914.201, 4722.917, 4540.079, 4365.273, 4198.108, 4038.212, 3885.235, 3738.844,
+                    3598.724, 3464.577, 3336.120, 3213.084, 3095.215, 2982.272, 2874.025, 2770.257, 2670.762, 2575.344,
+                    2483.816, 2396.003, 2311.736, 2230.855, 2153.209, 2078.653, 2007.050, 1938.270, 1872.188, 1808.687,
+                    1747.653, 1688.980, 1632.565, 1578.312, 1526.127, 1475.923, 1427.615, 1381.124, 1336.372, 1293.288,
+                    1251.801, 1211.845, 1173.357, 1136.277, 1100.546, 1066.109, 1032.915, 1000.912, 970.054, 940.293,
+                    911.586, 883.891, 857.168, 831.379, 806.487, 782.457, 759.255, 736.850, 715.210, 694.306, 674.111,
+                    654.596, 635.736, 617.507, 599.884, 582.845, 566.368, 550.432, 535.018, 520.105, 505.675, 491.712,
+                    478.197, 465.115, 452.449, 440.186, 428.310, 416.807, 405.665, 394.870, 384.411, 374.275, 364.452,
+                    354.930, 345.698, 336.748, 328.068, 319.651, 311.486, 303.566, 295.881, 288.425, 281.188, 274.165,
+                    267.348, 260.729, 254.303, 248.063, 242.003, 236.117, 230.400, 224.845, 219.448, 214.203, 209.106,
+                    204.152, 199.336, 194.654, 190.102, 185.675, 181.370};
+
+    int first = 0;

```

Comment:
Binary search looks good but should be A) in its own static local helper function, and B) dont use "magic numbers". Use the size of the array minus one for last and define the 54.0 offset in a #define with the LUT.

And of course, per the first comment, this should be run on every tm_reading index of the TempModel_t and the result stored in temps.